### ISSUE TITLE
Windowing instead of pagination

### ISSUE THREAD

**Issue by neverfox** [Wildhoney/Snapshot.js#26]
I was hoping to be able to implement lazy-loading scrolling instead of pagination using Snapshot.js, but it appears to be limited to that paradigm, possibly for good reasons. But is it currently possible to do that or would that be an enhancement that would have to be developed? I would think that if it could accept something like `startingIndex` instead of `pageNumber`, it could work.

**Comment by Wildhoney** [comment]
Aren't the two paradigms very similar &mdash; at least from the perspective of the server-side implementation? Instead of having next and previous, you could scroll, and simply keep loading the additional pages. Use the `perPage` to limit how many each scroll would yield.

**Comment by neverfox** [comment]
Hmm, you might be right. I was thinking mainly in terms of the client-side differences. For some reason, I got the impression that each new request to the server altered the whole collection landscape server-side, but I could choose the necessary merging strategy to keep my window populated appropriately (f the scrolling stops mid-way between pages for example, or if I'm buffering). Is that what you're getting at?

**Comment by Wildhoney** [comment]
Yep &mdash; you're right in that it alters the collection landscape, but that's more of a client-side problem.

I suppose taking the user to the top of the page to begin their infinite scroll journey from scratch would be the most logical. That's how most infinite scrolls that I've seen operate, because otherwise where you are on the page *post* filter/reorder would bear no resemblance to where you were *pre* filter/reorder &mdash; and thus simply confuse the user. There would also be DOM issues (unless you're being clever by not rendering unseen pages), such as if the user had scrolled to page 5001, then repainting all of those pages when reordering would be terribly slow.

As long as you don't reorder and/or apply filters, then each request is *predictable*.

**Comment by neverfox** [comment]
> I suppose taking the user to the top of the page to begin their infinite scroll journey from scratch would be the most logical.

Right. Filtering when only viewing some of the data screws with one's mental model, but I'm not sure if it's any worse for infinite scrolling than the pagination story, since most of keeps you from getting disoriented is some kind of feedback about where you are whether it's (page 3 of 10,000) or (records 100-110 of 100,000).

> unless you're being clever by not rendering unseen pages

Yes, I am indeed, using [react-list](https://github.com/orgsync/react-list).

In any case, I'm going to give it a shot and I can report back about how it went.
